src/d/actor/d_a_ks.cpp

```diff
@@ -5,125 +5,1613 @@
 
 #include "d/actor/d_a_ks.h"
 #include "d/d_procname.h"
+#include "d/d_com_inf_game.h"
+#include "d/d_s_play.h"
+#include "f_op/f_op_camera.h"
+#include "d/d_snap.h"
+#include "d/actor/d_a_player_main.h"
+#include "d/actor/d_a_gm.h"
+#include "d/res/res_ks.h"
+
+#include "weak_bss_936_to_1036.h" // IWYU pragma: keep
+#include "weak_data_1811.h" // IWYU pragma: keep
+
+int KS_ALL_COUNT;
+int KUTTUKU_ALL_COUNT;
+BOOL HEAVY_IN;
+int GORON_COUNT;
 
 /* 00000078-000002CC       .text draw_SUB__FP8ks_class */
-void draw_SUB(ks_class*) {
-    /* Nonmatching */
+void draw_SUB(ks_class* i_this) {
+    J3DModel* pBodyModel = i_this->mpBodyMorf->getModel();
+    J3DModel* pEyeModel = i_this->mpEyeMorf->getModel();
+
+    cXyz local_24 = dComIfGp_getCamera(0)->mLookat.mEye - i_this->current.pos;
+
+    int iVar3 = cM_atan2s(local_24.x, local_24.z);
+    int iVar4 = (s16)-cM_atan2s(local_24.y, std::sqrtf(local_24.x * local_24.x + local_24.z * local_24.z));
+    
+    f32 fVar1 = 0.0f;
+    if (i_this->m2D0) {
+        fVar1 = i_this->m304;
+    }
+
+    mDoMtx_trans(mDoMtx_stack_c::get(), i_this->current.pos.x, i_this->current.pos.y + fVar1, i_this->current.pos.z);
+    
+    mDoMtx_YrotM(mDoMtx_stack_c::get(), iVar3);
+    mDoMtx_XrotM(mDoMtx_stack_c::get(), iVar4);
+    mDoMtx_ZrotM(mDoMtx_stack_c::get(), i_this->shape_angle.z + i_this->m2FA);
```

Comment:
```suggestion
    mDoMtx_stack_c::transS(i_this->current.pos.x, i_this->current.pos.y + fVar1, i_this->current.pos.z);
    
    mDoMtx_stack_c::YrotM(iVar3);
    mDoMtx_stack_c::XrotM(iVar4);
    mDoMtx_stack_c::ZrotM(i_this->shape_angle.z + i_this->m2FA);
```
No practical difference, but the devs use these more specific inlines as they're shorter than typing `mDoMtx_stack_c::get()` everywhere

Also applies a few lines down from here